tests/components/solaredge_modules/test_coordinator.py

```diff
@@ -0,0 +1,212 @@
+"""Tests for the SolarEdge Modules coordinator."""
+
+from datetime import datetime
+from unittest.mock import AsyncMock
+
+import pytest
+from solaredge_web import EnergyData
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.recorder import Recorder
+from homeassistant.components.recorder.statistics import statistics_during_period
+from homeassistant.components.solaredge_modules.coordinator import (
+    SolarEdgeModulesCoordinator,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.util import dt as dt_util
+
+from tests.common import MockConfigEntry
+from tests.components.recorder.common import async_wait_recording_done
+
+
+async def test_coordinator_first_run(
+    recorder_mock: Recorder,
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_solar_edge_web: AsyncMock,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test the coordinator on its first run with no existing statistics."""
+    coordinator = SolarEdgeModulesCoordinator(hass, mock_config_entry)
+    await coordinator._async_update_data()
+
+    await async_wait_recording_done(hass)
+    stats = await hass.async_add_executor_job(
+        statistics_during_period,
+        hass,
+        dt_util.as_utc(datetime(1970, 1, 1, 0, 0)),
+        None,
+        {
+            "solaredge_modules:123456_1001",
+            "solaredge_modules:123456_1002",
+        },
+        "hour",
+        None,
+        {"state", "sum"},
+    )
+    assert stats == snapshot
+
+
+async def test_coordinator_subsequent_run(
+    recorder_mock: Recorder,
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_solar_edge_web: AsyncMock,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test the coordinator correctly updates statistics on subsequent runs."""
+    mock_solar_edge_web.async_get_equipment.return_value = {
+        1001: {"displayName": "1.1"},
+    }
+
+    coordinator = SolarEdgeModulesCoordinator(hass, mock_config_entry)
+
+    # Run the coordinator for the first time to create initial statistics
+    await coordinator._async_update_data()
+    await async_wait_recording_done(hass)
+
```

Comment:
Instead, should we just have HA set up the integration and freeze and skip time?

Reply:
Done. But test_coordinator_subsequent_run_with_gap became flaky failing ~3%. I had to patch _async_update_data to signal completion via an event, see last commit "fix flakiness".